gusto/state.py

```diff
@@ -249,9 +251,19 @@ def __init__(self, mesh, vertical_degree=None, horizontal_degree=1,
             if dim == 2:
                 self.perp = lambda u: as_vector([-u[1], u[0]])
 
+        # project test function for hydrostatic case
+        if self.h:
+            self.P = lambda u: u - self.k*inner(u, self.k)
+        else:
+            self.P = lambda u: u
+
         #  Constant to hold current time
         self.t = Constant(0.0)
 
+    def parameter_update(self):
+        # method to update parameters if needed
```

Comment:
This is not how you write a docstring, you write a docstring like:
```python
def parameter_update(self):
    """Do foo to the parameters.

More long-winded explanation here."""
    implementation_here
```

In particular, this comment gives me no indication as to which parameters should be updated, when it might be needed, or anything else.

Reply:
Sorry for the bad docstring, how about 
```
def parameter_update(self, t):
       """
       Update parameters such as off-centring parameter,
       dumpfrequency at specific times during model run
        
       :arg t: the current model time
       """
       pass
```
with ``` state.parameter_update(t)``` in timeloop and 
```
def implicit_first_step(time):
    if time == dt:
        state.timestepping.alpha = 0.5
```
in the example file? I'll also rewrite the current docstrings of ```setup_diagnostics``` and ```setup_dump```.

Reply:
See my comment in the test file about not closing over state.